backend/modules/parsers/multimodalparser.py

```diff
@@ -92,52 +108,41 @@ def __init__(
 
     async def call_vlm_agent(
         self,
+        llm: BaseChatModel,
         base64_image: str,
         page_number: int,
         prompt: str = "Describe the information present in the image in a structured format.",
     ):
-        logger.info(f"Processing Image... {page_number}")
-        headers = {
-            "Content-Type": "application/json",
-            "Authorization": f"Bearer {settings.TFY_API_KEY}",
-            # Set the tfy_log_request to "true" in X-TFY-METADATA header to log prompt and response for the request
-            "X-TFY-METADATA": json.dumps(
-                {"tfy_log_request": "true", "Custom-Metadata": "Custom-Value"}
-            ),
-        }
-        # Parallel calls to VLM
-        async with aiohttp.ClientSession() as session:
-            payload = {
-                "model": self.vision_model,
-                "messages": [
+        content = [
+            HumanMessage(
+                [
+                    {
+                        "type": "text",
+                        "text": prompt,
+                    },
                     {
-                        "role": "user",
-                        "content": [
-                            {"type": "text", "text": prompt},
-                            {
-                                "type": "image_url",
-                                "image_url": {
-                                    "url": f"data:image/jpeg;base64,{base64_image}"
-                                },
-                            },
-                        ],
-                    }
-                ],
-                "max_tokens": 2048,
+                        "type": "image_url",
+                        "image_url": {
+                            "url": f"data:image/jpeg;base64,{base64_image}",
```

Comment:
Are we sure, we'll always get jpeg?

Reply:
This works with png too. we just use base64 encoding of the image